libs/application-generic/src/services/resource-validator.service.ts

```diff
@@ -63,36 +82,31 @@ export class ResourceValidatorService {
   }
 
   private async getWorkflowLimit(environment: EnvironmentEntity, organization: OrganizationEntity) {
-    const maxWorkflowSystemLimit = await this.getMaxWorkflowSystemLimit(environment, organization);
+    const systemLimitMaxWorkflow = await this.getSystemLimitWorkflows(environment, organization);
+
+    // If the system limit is not the default, we need to use it as the absolute limit for special cases instead of the tier limit
+    const isSpecialLimit = systemLimitMaxWorkflow !== SYSTEM_LIMITS.WORKFLOWS;
```

Comment:
Nitpicky: Inverting the logic makes the code simpler
```suggestion
    const isDefaultSystemLimit = systemLimitMaxWorkflow === SYSTEM_LIMITS.WORKFLOWS;
```

Reply:
it is, but i does not tell the story about special cases